transport-native-unix-common/src/main/java/io/netty/channel/unix/IovArray.java

```diff
@@ -150,7 +152,14 @@ private boolean add(long memoryAddress, long addr, int len) {
         size += len;
         ++count;
 
-        if (ADDRESS_SIZE == 8) {
+        if (ADDRESS_SIZE == 16) {
+            if (PlatformDependent.hasUnsafe()) {
+                PlatformDependent.putObject(memoryAddress.add(baseOffset), 0, addr);
```

Comment:
might be a version where you pass addr, offset? Rather than making two native calls. Same for the putLong below.